core/src/main/java/fr/sncf/osrd/infra/implementation/tracks/undirected/UndirectedInfraBuilder.java

```diff
@@ -168,13 +171,62 @@ private TrackSectionImpl makeTrackSection(RJSTrackSection track) {
                 track.id,
                 ImmutableSet.copyOf(operationalPointsPerTrack.get(track.id)),
                 track.geo,
-                track.sch
+                track.sch,
+                buildLoadingGaugeLimits(track.loadingGaugeLimits)
         );
         builder.addEdge(begin, end, edge);
         edge.gradients = makeGradients(track);
         return edge;
     }
 
+    /** Builds the ranges of blocked loading gauge types on the track */
+    private ImmutableRangeMap<Double, ImmutableSet<RJSLoadingGaugeType>> buildLoadingGaugeLimits(
```

Comment:
can't it be static?

Reply:
It calls `getCompatibleGaugeTypes`, which needs the `warningRecorder`